### DISCUSSION TITLE
Welcome to grott-home-assistant-add-on Discussions!

### DISCUSSION THREAD

**Discussion by egguy** [egguy/grott-home-assistant-add-on#4] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by kungfuko** [comment]
Hi

I have followed your guide, and both Mosquitto broker and Grott - Home Assistant native MQTT integration - current seems to be fetching data from my inverter. But i cant seem to find any entities. What can i be doing wrong?

**Reply by rklomp** [reply]
Did you figure out how to get groeatt entities?

**Reply by egguy** [reply]
Hello, you need to reconfigure the stick to point to the IP of your home assistant instance. Grott only works on push. It can't pull the data from the logger without a reconfiguration.
You can see in the log the appearance of messages in case of the configuration working successfully.

**Comment by knetsooj** [comment]
Hi Etienne, first of thank you for providing this addon! 

It works perfect to read out data in my hass, however would it be possible to send commands as well? i saw the grott server now supports an API as of version 2.7.8 but i'm not sure where to start if i would like to enable/ trigger this from home assistant

thanks in advance

**Comment by teo693** [comment]
Hello, is there any plan to introduce a  [server mod](https://github.com/johanmeijer/grott/discussions/98) to your add-on?

**Reply by egguy** [reply]
I took a look at it. It would be possible to create something. I will try in the next iteration of the add-on.

**Reply by teo693** [reply]
Great, I'm looking forward to testing it out. Just let me know when you have an update on your progress.

**Comment by pi4lab** [comment]
**Growatt model MID12KTL3-XL**

Just want to tell you that this (3 Phase) model works like a charm with this add-on with 'tl3" type setup. 
Receives local updates every 60 secs directly from inverter and  Growatt Web and ShinePhone services are unaffected. 

It takes some time for the data to arrive in MQTT after setup/change of Shine-Lan/Wifi but my data started to flow in MQTT at next wakeup (Next Morning). Proxy service was in service with immediate effect.

![Mb-MQTT](https://github.com/egguy/grott-home-assistant-add-on/assets/99322979/0161776d-deb2-494d-8fd0-db83bf9b0e1e)

![Gwatt-sensors](https://github.com/egguy/grott-home-assistant-add-on/assets/99322979/04c3edab-8dca-49e0-a51e-f7d569b18b3c)

**Reply by nh2** [reply]
> takes some time for the data to arrive in MQTT after setup/change of Shine-Lan/Wifi but my data started to flow in MQTT at next wakeup (Next Morning).

I think the device does not react to changes in the configured target IP until it restarts. It seems to self-restart daily, thus your "next morning".

If you restart the device manually after changing the settings, the data should show up immediately.

To restart my `Growatt Shinelink/-X`, I go to `Network Settings`, set the option box `Reboot Able` to `Yes` and click `Save`:

<img width="937" height="529" alt="image" src="https://github.com/user-attachments/assets/1b9ae6b1-3b79-47e5-8399-6a56fe6cf929" />

**Comment by sidequest-tomas** [comment]
This is great. Growatt server is very unreliable, now the data seem to be flowing nice to my HA :)

Does anyone have a nice GUI setup of values to share or other tips to get this visually nice?

**Comment by arjg1979** [comment]
Thank you so much. Finally got rid of the unreliable cloud Growatt integration. It was so easy to configure thanks to your documentation.

**Comment by JWKoppie** [comment]
Hi, 
I'm a bit of a newbie with Home Assistent and used the default Growatt Addon of muppet3000.'
But now this add-on isn't working anymore I stumbled in on your add-on which almost works great
I can read the Growatt inverter MIN 5000TL-XH and used MQTT Explorer to see what settings are used.

The default that is used is not working for me because the 'state class' option has a value 'total'
The result is that every morning in my Energy dashboard I see a huge bar at 0:00-1:00 with the total energy
If I change the value from 'total' to 'total_increasing' and publish this setting and retain it with MQTT Explorer, the next day it is working fine

Below the default of what I see with MQTT Explorer
```
{
  "name": "Solar PV1 energy today",
  "unique_id": "grott_YBK4CFJWHO_epv1today",
  "state_topic": "homeassistant/grott/YBK4CFJWHO/state",
  "device": {
    "identifiers": [
      "YBK4CFJWHO"
    ],
    "name": "YBK4CFJWHO",
    "manufacturer": "GrowWatt"
  },
  "icon": "mdi:solar-power",
  "state_class": "total",
  "device_class": "energy",
  "unit_of_measurement": "kWh",
  "value_template": "{{ value_json.epv1today | float / 10 }}"
}

```
But.. everytime I reboot my device (a NUC) the value is changed back to: "state_class": "total",
And.... when changing from one month to another month, the same happens.

How (and where) do configure my setting to "state_class": "total_increasing", a let this value be the standard value, even when I reboot my system or after a change of month.

Just restarting HA is no problem, than the changed setting is still there.

**Reply by egguy** [reply]
Hi,

You should only use the the lifetime / total sensor.
HA automatically manage the day change with the total se sensor. 

It's suggested by HA to use the today sensor, but this is an error. I can't find a way to prevent them to be displayed.

**Comment by JWKoppie** [comment]
But.... I have two separate strings of panels (East and West side of my roof), The lifetime/total sensor displays these two strings combined.
With the PV1 today and PV2 today sensors I can see what these two strings separate from each other are doing. Which is very usefull especially to see if both strings are working. So the lifetime/total sensor is not the best option for me. 
Is there maybe a possibilty that I can use a script that can revert the state if it is changed to 'total' after a reboot?

**Reply by egguy** [reply]
There is also the `Solar PV1 production (Total)` and `Solar PV2 production (Total)` in addition of the `"Lifetime solar energy"` who should be each of your string.
I won't recommend this and would prefer to use the total, you have the Customise entity, who should be permanent: https://www.home-assistant.io/docs/configuration/customizing-devices/

**Reply by JWKoppie** [reply]
For now I'm overriding the MQTT settings in my configuration.yaml with the following

```
mqtt:
  - sensor:
    - name: Solar PV1 energy today
      unique_id: grott_YBK4CFJWHO_epv1today
      device_class: 'energy'
      unit_of_measurement: 'kWh'
      state_class: 'total_increasing'
      icon: 'mdi:solar-power'
      state_topic: 'homeassistant/grott/YBK4CFJWHO/state'
      value_template: '{{ value_json.epv1today | float / 10 }}'
  - sensor:
    - name: Solar PV2 energy today
      unique_id: grott_YBK4CFJWHO_epv2today
      device_class: 'energy'
      unit_of_measurement: 'kWh'
      state_class: 'total_increasing'
      icon: 'mdi:solar-power'
      state_topic: 'homeassistant/grott/YBK4CFJWHO/state'
      value_template: '{{ value_json.epv2today | float / 10 }}'


```
This seems to work. After a reboot of my NUC, or just a restart of HA the settings I want are still there. 
Just wait to see what happens at the end of the month.
